Plugins/Statistics/GitImpact/FormImpact.cs

```diff
@@ -30,12 +30,7 @@ protected override void OnClosed(EventArgs e)
 
         private void Impact_Invalidated(object sender, InvalidateEventArgs e)
         {
-            ThreadHelper.JoinableTaskFactory.Run(
-                async () =>
-                {
-                    await this.SwitchToMainThreadAsync();
-                    UpdateAuthorInfo(Impact.GetSelectedAuthor());
-                });
+            Invoke(() => UpdateAuthorInfo(Impact.GetSelectedAuthor()));
```

Comment:
❗ `JoinableTaskFactory.Run` should not be wrapped in helper methods. This one should be returned to the original form.

Reply:
Please have a look at the implementation of `Control.Invoke()`. It ensures that the control still exists.

Reply:
For GitExtensions this is handled in `SwitchToMainThreadAsync`:
https://github.com/gitextensions/gitextensions/blob/79a312b929947bce9a3ce5f3e2096c11f543ad31/GitExtUtils/GitUI/ControlThreadingExtensions.cs#L46-L68

Reply:
Thank you. I mistook `Control.Invoke` for `GitUIExtensions.InvokeSync`.
I have replaced all usages of `Control.Invoke`. There were a few others.

Reply:
Note that `InvokeSync` should also not be used. The only form here which adheres to the vs-threading best practices is the form which this code had before making the change.

Reply:
The coding practices for vs-threading were designed around "everyone is doing the same thing" so transitions to (and dependencies on) the main thread look the same in pretty much any code or library using it. The additional lines add clarity and make it more likely that threading-related analyzers will identify if problems occur.

Reply:
I am not convinced. I thought the times were gone where developers served tools and not vice versa.
Further discussions are not worth it. I give up with this final complaint.
I have reverted all new usages of `InvokeSync` from this PR.